apps/coscrad-frontend/src/utils/generic-components/presenters/bilingual-multilingual-text-presenter.tsx

```diff
@@ -0,0 +1,74 @@
+import { IMultilingualText } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { Box } from '@mui/material';
+import { useContext } from 'react';
+import { BilingualTextPresenter } from '../../../components/resources/utils/bilingual-text-presenter';
+import { ConfigurableContentContext } from '../../../configurable-front-matter/configurable-content-provider';
+import { FlatMultilingualTextPresenter } from './flat-multilingual-text-presenter';
+import { isInLanguage } from './is-in-language';
+import { isOriginalTextItem } from './is-original-text-item';
+
+export interface BilingualMultilingualTextPresenterProps {
+    text: IMultilingualText;
+}
+
+export const BilingualMultilingualTextPresenter = ({
+    text,
+}: BilingualMultilingualTextPresenterProps): JSX.Element => {
+    const { defaultLanguageCode } = useContext(ConfigurableContentContext);
+
+    const textItemWithDefaultLanguage = text.items.find((item) =>
+        isInLanguage(defaultLanguageCode, item)
+    );
+
+    const primaryMultilingualTextItem = isNullOrUndefined(textItemWithDefaultLanguage)
+        ? text.items.find((item) => isOriginalTextItem(item))
+        : textItemWithDefaultLanguage;
+
+    const { languageCode: languageCodeOfPrimaryTextItem } = primaryMultilingualTextItem;
+
+    const translations = text.items.filter(
+        (item) => !isInLanguage(languageCodeOfPrimaryTextItem, item)
+    );
+
+    const isTranslated: boolean = translations.length > 0 ? true : false;
+
+    return (
+        <Box width={'fit-content'} data-testid="multilingual-text-display">
+            {isTranslated ? (
+                // TODO clean this up
+                /**
+                 * Note that we are prioritizing a translation item is in the default
+                 * language code or English as this will almost always be the case
+                 * in the data. Otherwise, we just pick the first translation. For now,
+                 * we aren't doing N-lingual sites with n>2, but if we ever move to this,
+                 * we need to update our logic here to be more robust to the fully
+                 * multilingual case.
+                 */
+                <BilingualTextPresenter
+                    textInPrimaryLanguage={primaryMultilingualTextItem.text}
+                    textInSecondaryLanguage={
+                        (
+                            translations.find(
+                                ({ languageCode }) => languageCode === defaultLanguageCode
+                            ) || translations[0]
+                        ).text
+                    }
+                />
+            ) : (
+                // formatBilingualText(
```

Comment:
You can remove this. If it's not used, remove the helper `formatBilingualText` altogether. Also, add a unit test for `BilingualMultilingualTextPresenter` since it has heavy logic. If we reuse the search logic for multilngual text elsewhere, we might want a unit-tested SST for this in a util. It might be interesting to break out the real `MultilingualText` into a shared `@cosrad/text-processing` lib so the front-end can just hydrate instances and use the methods.

Reply:
Good call. Let's add a story to sort this out.